src/common/redis_module/ray_redis_module.cc

```diff
@@ -963,6 +963,88 @@ bool is_nil(const std::string &data) {
   return true;
 }
 
+/// Clean all the objects/actors/tasks/fucntions under one batch.
+///
+/// This is called from a client with the command:
+//
+///    RAY.BATCH_CLEAN <table_prefix> <pubsub_channel> <batch_id>
+///
+/// \param table_prefix The prefix string for keys in batch table.
+/// \param pubsub_channel The pubsub channel name that notifications for
+///        this key should be published to.
+/// \param batch_id The ID of the batch to delete.
+/// \return OK.
+int BatchClean_RedisCommand(RedisModuleCtx *ctx,
+                            RedisModuleString **argv,
+                            int argc) {
+  RedisModule_AutoMemory(ctx);
+
+  if (argc < 4) {
+    return RedisModule_WrongArity(ctx);
+  }
+
+  RedisModuleString *prefix_str = argv[1];
+  RedisModuleString *pubsub_channel_str = argv[2];
+  RedisModuleString *batch_id = argv[3];
+
+  // Publish this batch clean message before cleaning this table.
+  // Read the table content to form the publish message.
+  flatbuffers::FlatBufferBuilder fbb;
+  RedisModuleKey *table_key_read =
+      OpenPrefixedKey(ctx, prefix_str, batch_id, REDISMODULE_READ);
+  if (table_key_read == nullptr) {
+    // This is an empty key, no need to process.
+    RedisModule_ReplyWithNull(ctx);
+    return REDISMODULE_OK;
+  } else {
+    TableEntryToFlatbuf(table_key_read, batch_id, fbb);
+  }
+
+  TablePubsub pubsub_channel = ParseTablePubsub(pubsub_channel_str);
+  if (pubsub_channel != TablePubsub::NO_PUBLISH) {
+    RedisModuleCallReply *reply =
+        RedisModule_Call(ctx, "PUBLISH", "sb", pubsub_channel_str,
+                         fbb.GetBufferPointer(), fbb.GetSize());
+    if (reply == nullptr) {
+      return RedisModule_ReplyWithError(ctx, "error during PUBLISH");
+    }
+  }
+  RedisModule_CloseKey(table_key_read);
```

Comment:
I guess auto memory management is enabled (so this line is not necessary) given no one calls this. You can double check it.

Reply:
Later in this function I will open this key again and delete it. So I think it is better to close the key first.